### DISCUSSION TITLE
Optimized method for meshing a large number of spheres

### DISCUSSION THREAD

**Discussion by Ammarkhat** [libfive/libfive#559] [Q&A]
Any recommendation on how to get a faster performance when meshing the union of a large number of spheres (in hundreds)? 

My code is in C++, and I am currently doing: 
`Tree out = sphere(2.51, -25.26, 22.72, 22.67);
out = min(out, sphere(6.85, -25.81, 18.41, 23.85));
out = min(out, sphere(6.67, -25.85, 18.58, 23.79));
out = min(out, sphere(6.96, -25.79, 18.30, 23.89));
out = min(out, sphere(6.67, -25.95, 18.54, 23.79));
out = min(out, sphere(6.99, -25.78, 18.27, 23.90));
out = min(out, sphere(6.67, -26.05, 18.46, 23.80));
out = min(out, sphere(6.97, -25.81, 18.27, 23.89));
out = min(out, sphere(6.58, -26.18, 18.41, 23.79));
...`
where sphere calls the standard API sphere function. 
Currently, it takes about 7 seconds for 170 spheres, and time grows with the number of spheres. I am wondering if there's an optimization option or a trick to make it faster?

**Comment by tbwilkinson** [comment]
I'm working on a similar problem at the moment, but for a different primitive SDF. I'm currently attempting to add a spatial acceleration structure (like a bounding volume hierarchy, BVH) and then when you evaluate a given point, it first finds the closest objects (or closest n objects) and then calculates the distance. At least this is what should happen in theory.

Have you managed to get any further in the time since you commented?
